packages/zoe/src/contracts/loan/types.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @typedef {AsyncIterable<undefined>} PeriodAsyncIterable
+ *
+ *  The asyncIterable used for notifications that a period has passed,
+ *  on which compound interest will be calculated using the
+ *  interestRate.
+ */
+
+/**
+ * @typedef {number} MMR
+ *  The Maintenance Margin Requirement, in percent. The default is
+ *  150, meaning that collateral should be worth at least 150% of the
+ *  loan. If the value of the collateral drops below mmr, liquidation
+ *  occurs.
+ */
+
+/**
+ * @typedef {Instance} AutoswapInstance
+ *   The running contract instance for an Autoswap or Multipool
+ *   Autoswap installation.  The publicFacet from the Autoswap
+ *   instance is used for producing an invitation to sell the
+ *   collateral on liquidation.
+ */
+
+/** @typedef {number} InterestRate
+ *
+ *   The rate in basis points that will be multiplied with the debt on
+ *   every period to compound interest.
+ */
+
+/**
+ * @typedef LoanTerms
+ *
+ * @property {MMR} [mmr=150]
+ *
+ * @property {AutoswapInstance} autoswapInstance
+ *
+ * @property {PriceAuthority} priceAuthority
+ *
+ *   Used for getting the current value of collateral and setting
+ *   liquidation triggers.
+ *
+ * @property {PeriodAsyncIterable} periodAsyncIterable
+ *
+ * @property {InterestRate} interestRate
+ */
+
+/**
+ * @typedef LenderSeatProperty
+ * @property {ZCFSeat} lenderSeat
+ *
+ *   The ZCFSeat representing the lender's position in the contract.
+ */
+
+/**
+ * @typedef {LoanTerms & LenderSeatProperty} LoanConfigWithLender
+ *
+ * The loan now has a lenderSeat, which is added to the config.
```

Comment:
I'm not sure that my correction is right, but I wanted to illustrate that it should say when it takes this value, rather than presuming that "now" will be meaningful.
```suggestion
 * After the lender exercises their invitation, this config adds the lenderSeat to the terms.
```

Reply:
Looks good, thanks for this edit!

Reply:
Oh, it does not add it to the terms. Those stay the same. It adds it to the config